# E-Commerce Sales & Customer Analytics
## 01 — Data Exploration & Data Quality Analysis

This notebook explores the structure, quality, consistency, and business characteristics of the e-commerce dataset.

### Objectives
- Understand the dataset structure
- Identify missing values and duplicates
- Validate important business calculations
- Investigate outliers
- Analyze negative-profit orders
- Analyze order status and returns
- Validate customer signup and order dates
- Document data-quality issues and decisions

## 1. Import Libraries

In [1]:
import pandas as pd

## 2. Load Dataset

In [2]:
df = pd.read_csv("../data/Ecommerce_Sales_Customer_Analytics_Dataset.csv")

## 3. Basic Dataset Structure

In [3]:
df.head()

,Order_ID,Order_Date,Customer_ID,Customer_Name,Customer_Email,Gender,Age,Customer_Signup_Date,Acquisition_Channel,State,...,Profit,Order_Status,Return_Flag,Delivery_Days,Payment_Method,Shipping_Method,Sales_Channel,Device_Type,Coupon_Code,Customer_Rating
0,ORD900000,2026-02-01,CUST115062,Customer_115062,customer115062@example.com,Male,18,2022-04-11,Organic Search,Chhattisgarh,...,1662.11,Shipped,No,NaN,Debit Card,Standard,Website,Mobile,APP10,3.5
1,ORD900001,2025-10-04,CUST113588,Customer_113588,customer113588@example.com,Male,36,2024-11-27,Social Media,Punjab,...,32912.66,Delivered,No,5.0,Debit Card,Standard,Mobile App,Mobile,No Coupon,3.9
2,ORD900002,2025-02-21,CUST115653,Customer_115653,customer115653@example.com,Female,39,2024-11-13,Paid Search,Odisha,...,224.55,Delivered,No,6.0,UPI,Same Day,Mobile App,Desktop,No Coupon,4.2
3,ORD900003,2026-02-13,CUST124908,Customer_124908,customer124908@example.com,Male,26,2023-03-13,Referral,West Bengal,...,742.87,Delivered,No,3.0,Net Banking,Standard,Mobile App,Desktop,No Coupon,5.0
4,ORD900004,2026-08-01,CUST113480,Customer_113480,customer113480@example.com,Female,18,2024-07-19,Social Media,Jharkhand,...,24258.86,Delivered,No,2.0,Credit Card,Standard,Website,Mobile,NEWUSER5,4.3


In [4]:
df.shape

(120000, 31)

## 4. Data Types & Dataset Information

In [5]:
df.columns

Index(['Order_ID', 'Order_Date', 'Customer_ID', 'Customer_Name',
       'Customer_Email', 'Gender', 'Age', 'Customer_Signup_Date',
       'Acquisition_Channel', 'State', 'City', 'Product_Category',
       'Product_Name', 'Quantity', 'Unit_Price', 'Discount_Percentage',
       'Discount_Amount', 'Gross_Sales', 'Net_Sales', 'Cost_of_Goods_Sold',
       'Shipping_Cost', 'Profit', 'Order_Status', 'Return_Flag',
       'Delivery_Days', 'Payment_Method', 'Shipping_Method', 'Sales_Channel',
       'Device_Type', 'Coupon_Code', 'Customer_Rating'],
      dtype='str')

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 120000 entries, 0 to 119999
Data columns (total 31 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Order_ID              120000 non-null  str    
 1   Order_Date            120000 non-null  str    
 2   Customer_ID           120000 non-null  str    
 3   Customer_Name         120000 non-null  str    
 4   Customer_Email        119026 non-null  str    
 5   Gender                120000 non-null  str    
 6   Age                   120000 non-null  int64  
 7   Customer_Signup_Date  120000 non-null  str    
 8   Acquisition_Channel   120000 non-null  str    
 9   State                 120000 non-null  str    
 10  City                  120000 non-null  str    
 11  Product_Category      120000 non-null  str    
 12  Product_Name          120000 non-null  str    
 13  Quantity              120000 non-null  int64  
 14  Unit_Price            120000 non-null  float64
 15  Discount_Pe

## 5. Missing Value Analysis

In [7]:
df.isnull().sum()

Order_ID                    0
Order_Date                  0
Customer_ID                 0
Customer_Name               0
Customer_Email            974
Gender                      0
Age                         0
Customer_Signup_Date        0
Acquisition_Channel         0
State                       0
City                        0
Product_Category            0
Product_Name                0
Quantity                    0
Unit_Price                  0
Discount_Percentage         0
Discount_Amount             0
Gross_Sales                 0
Net_Sales                   0
Cost_of_Goods_Sold          0
Shipping_Cost               0
Profit                      0
Order_Status                0
Return_Flag                 0
Delivery_Days           13256
Payment_Method              0
Shipping_Method             0
Sales_Channel               0
Device_Type                 0
Coupon_Code              1819
Customer_Rating          1453
dtype: int64

In [8]:
df[df["Delivery_Days"].isnull()]["Order_Status"].value_counts()

Order_Status
Cancelled     4860
Shipped       4720
Processing    3676
Name: count, dtype: int64

## 6. Duplicate Analysis

In [9]:
df.duplicated().sum()

np.int64(0)

In [10]:
df["Order_Date"].duplicated().sum()

np.int64(119392)

In [11]:
df["Order_ID"].nunique()

120000

In [12]:
df.groupby("Order_ID").size().describe()

count    120000.0
mean          1.0
std           0.0
min           1.0
25%           1.0
50%           1.0
75%           1.0
max           1.0
dtype: float64

## 7. Statistical Summary

In [13]:
df.describe()

,Age,Quantity,Unit_Price,Discount_Percentage,Discount_Amount,Gross_Sales,Net_Sales,Cost_of_Goods_Sold,Shipping_Cost,Profit,Delivery_Days,Customer_Rating
count,120000.000000,120000.000000,120000.000000,120000.000000,120000.000000,120000.000000,120000.000000,120000.000000,120000.000000,120000.000000,106744.000000,118547.000000
mean,32.216775,1.621133,14783.421951,9.149752,2202.321011,24021.952300,21819.631298,14629.988622,64.883089,7124.759570,4.135858,4.059531
std,8.558381,0.955956,20931.621026,5.682975,4740.810766,42012.358535,38262.887324,25960.074074,31.124362,13148.455442,2.066867,0.678763
min,18.000000,1.000000,80.010000,0.030000,0.310000,80.070000,67.830000,38.450000,25.000000,-135.380000,1.000000,1.000000
25%,26.000000,1.000000,1964.680000,4.830000,177.697500,2557.882500,2327.595000,1561.600000,42.570000,685.995000,2.000000,3.600000
50%,32.000000,1.000000,5123.035000,8.130000,565.505000,7360.345000,6666.390000,4422.015000,57.840000,2057.005000,4.000000,4.100000
75%,38.000000,2.000000,17459.200000,12.432500,1983.615000,24143.360000,22082.312500,14879.152500,77.132500,7234.327500,6.000000,4.600000
max,68.000000,5.000000,84999.850000,41.170000,132965.240000,424361.940000,416467.420000,313825.950000,179.990000,175019.080000,9.000000,5.000000


## 8. Business Rule Validation

### 8.1 Gross Sales Validation

In [14]:
df["Calculated_Gross_Sales"] = df["Unit_Price"] * df["Quantity"]

In [15]:
df["Gross_Sales_Difference"] = (
    df["Gross_Sales"] - df["Calculated_Gross_Sales"]
)

In [16]:
df["Gross_Sales_Difference"].abs().describe()

count    120000.000000
mean          0.002423
std           0.004690
min           0.000000
25%           0.000000
50%           0.000000
75%           0.000000
max           0.020000
Name: Gross_Sales_Difference, dtype: float64

In [17]:
df["Order_ID"].duplicated().sum()

np.int64(0)

### 8.2 Net Sales Validation

In [18]:
df["Calculated_Net_Sales"] = (
    df["Gross_Sales"] - df["Discount_Amount"]
)

In [19]:
df["Net_Sales_Difference"] = (
    df["Net_Sales"] - df["Calculated_Net_Sales"]
)

In [20]:
df["Net_Sales_Difference"].abs().describe()

count    1.200000e+05
mean     2.486417e-03
std      4.322274e-03
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      2.910383e-11
max      1.000000e-02
Name: Net_Sales_Difference, dtype: float64

### 8.3 Profit Validation

In [21]:
df["Calculated_Profit"] = (
    df["Net_Sales"]
    - df["Cost_of_Goods_Sold"]
    - df["Shipping_Cost"]
)

In [22]:
df["Profit_Difference"] = (
    df["Profit"] - df["Calculated_Profit"]
)

In [23]:
df["Profit_Difference"].abs().describe()

count    1.200000e+05
mean     3.323583e-03
std      4.710606e-03
min      0.000000e+00
25%      5.684342e-14
50%      9.094947e-13
75%      1.000000e-02
max      1.000000e-02
Name: Profit_Difference, dtype: float64

## 9. Outlier Analysis

### 9.1 Unit Price Outlier Analysis

In [24]:
Q1 = df["Unit_Price"].quantile(0.25)
Q3 = df["Unit_Price"].quantile(0.75)

IQR = Q3 - Q1

print("Q1:", Q1)
print("Q3:", Q3)
print("IQR:", IQR)

Q1: 1964.68
Q3: 17459.199999999997
IQR: 15494.519999999997


In [25]:
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

print("Lower Bound:", lower_bound)
print("Upper Bound:", upper_bound)

Lower Bound: -21277.099999999995
Upper Bound: 40700.979999999996


In [26]:
unit_price_outliers = df[
    (df["Unit_Price"] < lower_bound) |
    (df["Unit_Price"] > upper_bound)
]

print("Number of Unit Price Outliers:", len(unit_price_outliers))

Number of Unit Price Outliers: 15827


In [27]:
unit_price_outliers[
    [
        "Order_ID",
        "Product_Category",
        "Product_Name",
        "Unit_Price",
        "Quantity",
        "Gross_Sales",
        "Net_Sales",
        "Profit"
    ]
].sort_values("Unit_Price", ascending=False).head(10)

,Order_ID,Product_Category,Product_Name,Unit_Price,Quantity,Gross_Sales,Net_Sales,Profit
117937,ORD1017937,Electronics,Tablet,84999.85,1,84999.85,78234.00,33402.56
3782,ORD903782,Electronics,Laptop,84999.38,1,84999.38,80452.85,37642.20
11908,ORD911908,Electronics,Smartwatch,84995.81,2,169991.63,154092.22,58836.70
15182,ORD915182,Electronics,Smartphone,84995.30,4,339981.22,315518.83,96473.54
57575,ORD957575,Electronics,Laptop,84994.53,1,84994.53,79743.13,17929.50
45995,ORD945995,Electronics,Mouse,84992.78,1,84992.78,81129.86,15489.93
18578,ORD918578,Electronics,Bluetooth Speaker,84981.84,3,254945.53,247565.94,69307.35
111983,ORD1011983,Electronics,Headphones,84966.51,1,84966.51,83575.65,18871.74
4039,ORD904039,Electronics,Keyboard,84963.76,1,84963.76,72133.72,13588.79
81532,ORD981532,Electronics,Keyboard,84958.32,1,84958.32,78130.02,26537.60


In [28]:
df.groupby("Product_Category")["Unit_Price"].agg(
    ["count", "min", "mean", "median", "max"]
).sort_values("max", ascending=False)

,count,min,mean,median,max
Product_Category,,,,,
Electronics,30246,500.90,42838.402260,42834.670,84999.85
Home & Kitchen,19214,250.07,12614.248184,12592.885,24999.74
Sports,9559,302.08,6146.944862,6143.250,11998.93
Fashion,20350,301.70,4603.386225,4591.785,8999.74
Toys,9485,200.15,3094.214252,3086.940,5999.32
Beauty,10829,150.69,2576.713179,2580.800,4999.44
Books,9576,100.26,1304.086416,1312.015,2499.90
Grocery,10741,80.01,1282.265937,1282.580,2499.76


In [29]:
df.groupby(
    ["Product_Category", "Product_Name"]
)["Unit_Price"].agg(
    ["count", "min", "median", "max"]
).sort_values("max", ascending=False).head(20)

count     min     median       max
Product_Category Product_Name                                         
Electronics      Tablet              3429  507.19  42558.130  84999.85
                 Laptop              3352  508.43  43032.155  84999.38
                 Smartwatch          3415  515.03  43215.710  84995.81
                 Smartphone          3342  567.07  40778.005  84995.30
                 Mouse               3345  554.23  43218.480  84992.78
                 Bluetooth Speaker   3390  588.23  43075.055  84981.84
                 Headphones          3302  507.59  43490.800  84966.51
                 Keyboard            3356  500.90  43165.625  84963.76
                 Monitor             3315  523.88  42364.250  84957.43
Home & Kitchen   Water Purifier      2377  250.07  12431.710  24999.74
                 Air Fryer           2373  295.57  12405.900  24996.03
                 Vacuum Cleaner      2453  255.81  12620.420  24994.33
                 Cookware Set        2438  259.73  12962.065  24990.69
                 Mixer Grinder       2453  256.67  12427.940  24988.79
                 Bedsheet            2431  255.26  12465.840  24987.43
                 Coffee Maker        2402  257.47  12407.385  24986.83
                 Storage Box         2287  269.20  13087.440  24979.29
Sports           Fitness Band        1317  303.49   6086.910  11998.93
                 Football            1384  302.08   6362.905  11997.42
                 Cricket Bat         1401  306.66   6277.470  11993.53

## 10. Negative Profit Analysis

In [30]:
negative_profit = df[df["Profit"] < 0]

print("Negative Profit Orders:", len(negative_profit))
print("Percentage of Orders:", len(negative_profit) / len(df) * 100)

Negative Profit Orders: 1122
Percentage of Orders: 0.935


In [31]:
negative_profit.groupby("Product_Category")["Profit"].agg(
    ["count", "sum", "mean", "min"]
).sort_values("count", ascending=False)

,count,sum,mean,min
Product_Category,,,,
Grocery,464,-15165.36,-32.683966,-135.38
Books,372,-11760.07,-31.613091,-133.32
Beauty,133,-3843.61,-28.899323,-126.25
Toys,73,-2058.31,-28.196027,-111.96
Fashion,54,-1292.03,-23.926481,-100.50
Home & Kitchen,13,-350.38,-26.952308,-65.28
Sports,12,-362.36,-30.196667,-87.55
Electronics,1,-20.49,-20.490000,-20.49


In [32]:
negative_profit[
    [
        "Order_ID",
        "Product_Category",
        "Product_Name",
        "Net_Sales",
        "Cost_of_Goods_Sold",
        "Shipping_Cost",
        "Profit"
    ]
].sort_values("Profit").head(10)

,Order_ID,Product_Category,Product_Name,Net_Sales,Cost_of_Goods_Sold,Shipping_Cost,Profit
38942,ORD938942,Grocery,Cooking Oil,229.07,186.60,177.85,-135.38
106449,ORD1006449,Grocery,Rice,97.28,76.60,155.52,-134.84
79884,ORD979884,Books,Fiction Book,119.49,81.34,171.47,-133.32
113532,ORD1013532,Grocery,Dry Fruits,142.62,116.84,158.01,-132.22
72901,ORD972901,Grocery,Cooking Oil,163.46,117.96,177.65,-132.15
107988,ORD1007988,Grocery,Snacks,94.44,77.05,147.18,-129.80
110508,ORD1010508,Books,Fiction Book,197.12,151.26,173.58,-127.73
110619,ORD1010619,Beauty,Makeup Kit,213.93,166.93,173.25,-126.25
90976,ORD990976,Grocery,Rice,141.22,106.41,158.51,-123.71
77496,ORD977496,Books,Self Help Book,229.74,174.16,177.47,-121.88


## 11. Order Status & Return Analysis

In [33]:
df["Order_Status"].value_counts()

Order_Status
Delivered     98350
Returned       8394
Cancelled      4860
Shipped        4720
Processing     3676
Name: count, dtype: int64

In [34]:
df["Return_Flag"].value_counts()

Return_Flag
No     111606
Yes      8394
Name: count, dtype: int64

In [35]:
pd.crosstab(df["Order_Status"], df["Return_Flag"])

Return_Flag,No,Yes
Order_Status,,
Cancelled,4860,0
Delivered,98350,0
Processing,3676,0
Returned,0,8394
Shipped,4720,0


In [36]:
return_rate = (df["Return_Flag"] == "Yes").mean() * 100

print("Return Rate:", return_rate, "%")

Return Rate: 6.995 %


## 12. Date Analysis

In [37]:
print("Order Date Range:")
print(df["Order_Date"].min(), "to", df["Order_Date"].max())

print("\nCustomer Signup Date Range:")
print(df["Customer_Signup_Date"].min(), "to", df["Customer_Signup_Date"].max())

Order Date Range:
2025-01-01 to 2026-08-31

Customer Signup Date Range:
2022-01-01 to 2025-12-30


In [38]:
df["Order_Date"] = pd.to_datetime(df["Order_Date"])
df["Customer_Signup_Date"] = pd.to_datetime(df["Customer_Signup_Date"])

print(df[["Order_Date", "Customer_Signup_Date"]].dtypes)

Order_Date              datetime64[us]
Customer_Signup_Date    datetime64[us]
dtype: object


## 13. Customer Signup-Date Validation

In [39]:
invalid_signup_orders = df[
    df["Customer_Signup_Date"] > df["Order_Date"]
]

print("Orders before customer signup:", len(invalid_signup_orders))

Orders before customer signup: 9349


In [40]:
invalid_signup_orders[
    [
        "Order_ID",
        "Customer_ID",
        "Customer_Name",
        "Customer_Signup_Date",
        "Order_Date",
        "Product_Name",
        "Net_Sales"
    ]
].sort_values("Customer_Signup_Date", ascending=False).head(10)

,Order_ID,Customer_ID,Customer_Name,Customer_Signup_Date,Order_Date,Product_Name,Net_Sales
85867,ORD985867,CUST123653,Customer_123653,2025-12-30,2025-01-05,Rice,2341.96
116259,ORD1016259,CUST122639,Customer_122639,2025-12-30,2025-01-30,Building Blocks,5715.25
37132,ORD937132,CUST100795,Customer_100795,2025-12-30,2025-11-08,Water Purifier,22405.31
115153,ORD1015153,CUST106257,Customer_106257,2025-12-30,2025-09-21,Smartphone,79105.18
43931,ORD943931,CUST109832,Customer_109832,2025-12-30,2025-09-27,Laptop,13008.90
47361,ORD947361,CUST104144,Customer_104144,2025-12-30,2025-11-28,Sneakers,11167.79
119286,ORD1019286,CUST106257,Customer_106257,2025-12-30,2025-07-23,Sunscreen,1750.40
62110,ORD962110,CUST100795,Customer_100795,2025-12-30,2025-07-29,Air Fryer,7464.05
69463,ORD969463,CUST104144,Customer_104144,2025-12-30,2025-12-29,Doll,4851.73
92066,ORD992066,CUST123148,Customer_123148,2025-12-30,2025-11-01,Tablet,40184.93


In [41]:
invalid_signup_orders["Customer_Signup_Date"].max()

Timestamp('2025-12-30 00:00:00')

In [42]:
invalid_signup_orders["Customer_Signup_Date"].value_counts().head(10)

Customer_Signup_Date
2025-12-22    106
2025-12-12     85
2025-12-27     69
2025-11-28     68
2025-11-19     67
2025-12-01     67
2025-11-05     67
2025-10-13     66
2025-09-18     64
2025-11-20     64
Name: count, dtype: int64

In [43]:
invalid_signup_orders["Customer_ID"].nunique()

4316

In [44]:
customer_date_check = df.groupby("Customer_ID").agg(
    Signup_Date=("Customer_Signup_Date", "first"),
    First_Order_Date=("Order_Date", "min"),
    Order_Count=("Order_ID", "count")
)

customer_date_check["Signup_Before_First_Order"] = (
    customer_date_check["Signup_Date"] <=
    customer_date_check["First_Order_Date"]
)

customer_date_check.head(10)

,Signup_Date,First_Order_Date,Order_Count,Signup_Before_First_Order
Customer_ID,,,,
CUST100001,2022-08-09,2025-01-13,8,True
CUST100002,2022-09-18,2025-12-28,8,True
CUST100003,2022-06-15,2025-01-05,7,True
CUST100004,2022-08-11,2025-06-08,6,True
CUST100005,2023-09-19,2025-01-11,6,True
CUST100006,2022-01-02,2025-02-13,3,True
CUST100007,2022-07-06,2025-03-18,6,True
CUST100008,2022-12-31,2025-03-01,7,True
CUST100009,2022-01-01,2025-05-18,8,True


In [45]:
customer_date_check["Signup_Before_First_Order"].value_counts()

Signup_Before_First_Order
True     20478
False     4316
Name: count, dtype: int64

In [46]:
customer_date_check["Signup_Order_Difference_Days"] = (
    customer_date_check["Signup_Date"]
    - customer_date_check["First_Order_Date"]
).dt.days

customer_date_check[
    customer_date_check["Signup_Before_First_Order"] == False
]["Signup_Order_Difference_Days"].describe()

count    4316.000000
mean      145.511121
std        90.813094
min         1.000000
25%        68.000000
50%       138.000000
75%       215.000000
max       361.000000
Name: Signup_Order_Difference_Days, dtype: float64

In [47]:
customer_date_check[
    customer_date_check["Signup_Before_First_Order"] == False
].sort_values(
    "Signup_Order_Difference_Days",
    ascending=False
).head(10)

,Signup_Date,First_Order_Date,Order_Count,Signup_Before_First_Order,Signup_Order_Difference_Days
Customer_ID,,,,,
CUST121154,2025-12-28,2025-01-01,6,False,361
CUST122639,2025-12-30,2025-01-03,7,False,361
CUST107561,2025-12-29,2025-01-03,6,False,360
CUST119593,2025-12-28,2025-01-02,6,False,360
CUST100709,2025-12-28,2025-01-02,7,False,360
CUST103811,2025-12-28,2025-01-03,4,False,359
CUST110233,2025-12-30,2025-01-05,4,False,359
CUST123653,2025-12-30,2025-01-05,6,False,359
CUST113789,2025-12-29,2025-01-05,6,False,358


## 14. Final Data Quality Summary

In [48]:
print("===== DATA QUALITY SUMMARY =====")

print("Total Orders:", len(df))
print("Total Columns:", len(df.columns))
print("Duplicate Rows:", df.duplicated().sum())
print("Unique Order IDs:", df["Order_ID"].nunique())

print("\nMissing Values:")
print(df.isnull().sum()[df.isnull().sum() > 0])

print("\nNegative Profit Orders:", (df["Profit"] < 0).sum())
print("Return Rate:", (df["Return_Flag"] == "Yes").mean() * 100, "%")

print("\nInvalid Customer Signup Dates:", len(invalid_signup_orders))
print("Customers with Signup Date Issue:", 
      (~customer_date_check["Signup_Before_First_Order"]).sum())

print("\nUnit Price IQR Outliers:", len(unit_price_outliers))

===== DATA QUALITY SUMMARY =====
Total Orders: 120000
Total Columns: 37
Duplicate Rows: 0
Unique Order IDs: 120000

Missing Values:


Customer_Email       974
Delivery_Days      13256
Coupon_Code         1819
Customer_Rating     1453
dtype: int64

Negative Profit Orders: 1122
Return Rate: 6.995 %

Invalid Customer Signup Dates: 9349
Customers with Signup Date Issue: 4316

Unit Price IQR Outliers: 15827


## 15. Key Findings & Data Quality Decisions

### Dataset
- 120,000 orders
- 31 original dataset columns

### Missing Values
- Customer Email: 974
- Delivery Days: 13,256
- Coupon Code: 1,819
- Customer Rating: 1,453

### Data Integrity
- Duplicate rows: 0
- Duplicate Order IDs: 0

### Business Validation
- Gross Sales calculations validated
- Net Sales calculations validated
- Profit calculations validated

### Outliers
- 15,827 Unit Price records identified using the IQR method
- Outliers were investigated rather than automatically deleted
- High prices are concentrated in product categories such as Electronics

### Profitability
- 1,122 orders have negative profit
- Negative-profit orders represent approximately 0.94% of all orders
- Grocery and Books contribute the largest number of loss-making orders

### Returns
- 8,394 returned orders
- Return rate: approximately 7.0%
- Return Flag and Order Status are internally consistent

### Customer Date Quality
- 9,349 orders have an order date earlier than the recorded customer signup date
- 4,316 customers are affected
- Maximum signup/order discrepancy: 361 days
- Original values are retained for traceability

In [49]:
temporary_columns = [
    "Calculated_Gross_Sales",
    "Gross_Sales_Difference",
    "Calculated_Net_Sales",
    "Net_Sales_Difference",
    "Calculated_Profit",
    "Profit_Difference"
]

df.drop(columns=temporary_columns, inplace=True)

In [50]:
print("Final dataset shape:", df.shape)

Final dataset shape: (120000, 31)
